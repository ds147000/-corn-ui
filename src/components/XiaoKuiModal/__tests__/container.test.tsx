import React from '../container'
import { fireEvent, render, waitFor } from '@testing-library/react'
import XiaokuiModal from '../container'

describe('XiaokuiModal', () => {
  test('快照', async () => {
    const screen = render(<XiaokuiModal visible btn={[]} />)
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('类型快照', async () => {
    const screen = render(
      <>
        <XiaokuiModal type="active" visible btn={[]} />
        <XiaokuiModal type="default" visible btn={[]} />
        <XiaokuiModal type="guide" visible btn={[]} />
        <XiaokuiModal type="notify" visible btn={[]} />
        <XiaokuiModal type="upadte" visible btn={[]} />
      </>
    )
    await waitFor(() => expect(screen.container).toMatchSnapshot())
  })

  test('设置子节点', async () => {
    const screen = render(<XiaokuiModal visible btn={[]} >我是小葵弹窗</XiaokuiModal>)
    await waitFor(() => screen.getByText('我是小葵弹窗'))

  })

  test('设置按钮', async () => {
    const screen = render(
      <XiaokuiModal
        visible
        btn={[{ text: '炫耀一下' }, { text: '解锁更多', type: 'error', ghost: true }]}
      >
        我是小葵弹窗
      </XiaokuiModal>
    )

    await waitFor(() => screen.getByText('我是小葵弹窗'))
    await waitFor(() => screen.getByText('炫耀一下'))
    await waitFor(() => screen.getByText('解锁更多'))
  })

  test('点击按钮', async () => {
    const onClick = jest.fn()
    const screen = render(
      <XiaokuiModal
        visible
        btn={[{ text: '炫耀一下' }, { text: '解锁更多', type: 'error', ghost: true }]}
        onClick={onClick}
      >
        我是小葵弹窗
      </XiaokuiModal>
    )
    await waitFor(() => screen.getByText('我是小葵弹窗'))
    const btn1 = await waitFor(() => screen.getByText('炫耀一下'))
    const btn2 = await waitFor(() => screen.getByText('解锁更多'))
    fireEvent.click(btn1)
    fireEvent.click(btn2)
    await waitFor(() => expect(onClick).toHaveBeenCalledTimes(2))
    await waitFor(() => expect(onClick.mock.calls[0][1]).toBe(0))
    await waitFor(() => expect(onClick.mock.calls[1][1]).toBe(1))
  })
})
