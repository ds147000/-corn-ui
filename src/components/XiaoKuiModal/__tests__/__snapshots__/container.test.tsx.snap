// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XiaokuiModal 快照 1`] = `
<div>
  <div
    class="xrk-drawer-mask xrk-drawer-mask-show"
    data-testid="mask"
  />
  <div
    class="xrk-drawer-body xrk-drawer-center-show xrk-drawer-center"
    data-testid="body"
  >
    <div
      class="xrk-xiaokui-modal"
    >
      <div
        class="taro-img xrk-xiaokui-modal-head"
      >
        <img
          class="taro-img__mode-scaletofill"
          src="https://assets.xrkmm.cn/u/4000002499670412/661d0be5-8f27-4478-978c-7f59b87ee877.png"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-body"
      >
        <div
          class="xrk-xiaokui-modal-content"
        />
        <div
          class="xrk-xiaokui-modal-btns"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-close"
      />
    </div>
  </div>
</div>
`;

exports[`XiaokuiModal 类型快照 1`] = `
<div>
  <div
    class="xrk-drawer-mask xrk-drawer-mask-show"
    data-testid="mask"
  />
  <div
    class="xrk-drawer-body xrk-drawer-center-show xrk-drawer-center"
    data-testid="body"
  >
    <div
      class="xrk-xiaokui-modal"
    >
      <div
        class="taro-img xrk-xiaokui-modal-head"
      >
        <img
          class="taro-img__mode-scaletofill"
          src="https://assets.xrkmm.cn/u/4000002499670412/c0ba2f4a-4638-4101-8676-5681de2b1e5d.png"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-body"
      >
        <div
          class="xrk-xiaokui-modal-content"
        />
        <div
          class="xrk-xiaokui-modal-btns"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-close"
      />
    </div>
  </div>
  <div
    class="xrk-drawer-mask xrk-drawer-mask-show"
    data-testid="mask"
  />
  <div
    class="xrk-drawer-body xrk-drawer-center-show xrk-drawer-center"
    data-testid="body"
  >
    <div
      class="xrk-xiaokui-modal"
    >
      <div
        class="taro-img xrk-xiaokui-modal-head"
      >
        <img
          class="taro-img__mode-scaletofill"
          src="https://assets.xrkmm.cn/u/4000002499670412/661d0be5-8f27-4478-978c-7f59b87ee877.png"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-body"
      >
        <div
          class="xrk-xiaokui-modal-content"
        />
        <div
          class="xrk-xiaokui-modal-btns"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-close"
      />
    </div>
  </div>
  <div
    class="xrk-drawer-mask xrk-drawer-mask-show"
    data-testid="mask"
  />
  <div
    class="xrk-drawer-body xrk-drawer-center-show xrk-drawer-center"
    data-testid="body"
  >
    <div
      class="xrk-xiaokui-modal"
    >
      <div
        class="taro-img xrk-xiaokui-modal-head"
      >
        <img
          class="taro-img__mode-scaletofill"
          src="https://assets.xrkmm.cn/u/4000002499670412/50447be8-8b75-4613-b39c-8675ec0e2133.png"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-body"
      >
        <div
          class="xrk-xiaokui-modal-content"
        />
        <div
          class="xrk-xiaokui-modal-btns"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-close"
      />
    </div>
  </div>
  <div
    class="xrk-drawer-mask xrk-drawer-mask-show"
    data-testid="mask"
  />
  <div
    class="xrk-drawer-body xrk-drawer-center-show xrk-drawer-center"
    data-testid="body"
  >
    <div
      class="xrk-xiaokui-modal"
    >
      <div
        class="taro-img xrk-xiaokui-modal-head"
      >
        <img
          class="taro-img__mode-scaletofill"
          src="https://assets.xrkmm.cn/u/4000002499670412/db56b91e-fb14-4824-9141-63a85a306a3f.png"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-body"
      >
        <div
          class="xrk-xiaokui-modal-content"
        />
        <div
          class="xrk-xiaokui-modal-btns"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-close"
      />
    </div>
  </div>
  <div
    class="xrk-drawer-mask xrk-drawer-mask-show"
    data-testid="mask"
  />
  <div
    class="xrk-drawer-body xrk-drawer-center-show xrk-drawer-center"
    data-testid="body"
  >
    <div
      class="xrk-xiaokui-modal"
    >
      <div
        class="taro-img xrk-xiaokui-modal-head"
      >
        <img
          class="taro-img__mode-scaletofill"
          src="https://assets.xrkmm.cn/u/4000002499670412/333ab89b-95e9-48ae-88fe-f2b0e79e42cc.png"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-body"
      >
        <div
          class="xrk-xiaokui-modal-content"
        />
        <div
          class="xrk-xiaokui-modal-btns"
        />
      </div>
      <div
        class="xrk-xiaokui-modal-close"
      />
    </div>
  </div>
</div>
`;
